corehq/apps/users/dbaccessors.py

```diff
@@ -94,14 +123,47 @@ def get_commcare_users_by_filters(domain, user_filters, count_only=False):
     """
     if not any([user_filters.get('role_id', None), user_filters.get('search_string', None),
                 user_filters.get('location_id', None), count_only]):
-        return get_all_commcare_users_by_domain(domain)
+        if user_type == MOBILE_USER_TYPE:
+            return get_all_commcare_users_by_domain(domain)
+        if user_type == WEB_USER_TYPE:
+            return get_all_web_users_by_domain(domain)
 
-    query = _get_es_query(domain, user_filters)
+    query = _get_es_query(domain, user_type, user_filters)
 
     if count_only:
         return query.count()
     user_ids = query.scroll_ids()
-    return map(CommCareUser.wrap, iter_docs(CommCareUser.get_db(), user_ids))
+    return map(CouchUser.wrap_correctly, iter_docs(CommCareUser.get_db(), user_ids))
+
+
+def count_invitations_by_filters(domain, user_filters):
+    return _get_invitations_by_filters(domain, user_filters, count_only=True)
+
+
+def get_invitations_by_filters(domain, user_filters):
+    return _get_invitations_by_filters(domain, user_filters)
+
+
+def _get_invitations_by_filters(domain, user_filters, count_only=False):
+    """
+    Similar to _get_users_by_filters, but applites to invitations.
+
+    Applies "search_string" filter to the invitations' emails. This does not
+    support ES search syntax, it's just a case-insensitive substring search.
+    Ignores location filter.
```

Comment:
i think this last statement should be avoided/rephrased to "other filters ignored".

In fact just say about the filters, it simply considers. In case a new filter is added to user in future, this documentation will become invalid.

Reply:
Updated in 5377d650580